### PR TITLE
LPS-142095 Upgrade yarn.lock to prevent vulnerabilities

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
https://issues.liferay.com/browse/LPS-142095

I am upgrading yarn.lock in order to resolve the issue in LPS-142095.
Let me know if there are any questions or comments about this.

Thank you.

### CODE REVIEW

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
Hey @fortunatomaldonado, could you improve PR's title? 💔 

Also, could you run `npx yarn-deduplicate` inside `modules/` folder and if there are changes, could you squash with the current commit? Thanks <3

**Comment by fortunatomaldonado** [comment]
Hello @diegonvs, is the new title okay? Let me know and I will change it if need be.

I was able to run `npx yarn-deduplicate` and no changes were found.

Let me know if there is anything else needed.
Thanks for checking this!

**Comment by diegonvs** [comment]
> is the new title okay?

Yep 

> I was able to run npx yarn-deduplicate and no changes were found.

Great then! Thx!

**Review by diegonvs — APPROVED** [review]

**Comment by bryceosterhaus** [comment]
@fortunatomaldonado what were the steps you took to upgrade this package? Did you manually do this by hand or did you run a specific command?

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, I manually changed the version and ran yarn update. Is there a specific way to go about doing this?
Thanks.

**Comment by bryceosterhaus** [comment]
@fortunatomaldonado I think we may need to utilize yarn resolutions for issues like this to make sure yarn doesn't accidentally re-generate with the older version. I can take a look today

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, truly appreciate the help on this!

**Comment by bryceosterhaus** [comment]
@fortunatomaldonado I don't think I have push access to this branch, but I just checked locally and we should add a yarn resolution for this.

So in the root package.json, add 

```json
	"resolutions": {
		"ua-parser-js": "^0.7.30"
	},
```

And then go ahead and run `yarn` and `npx yarn-deduplicate yarn.lock`

The yarn.lock should then look something like this
Screen Shot 2021-11-10 at 11 53 34 AM

By adding the resolutions, we just ensure yarn knows to keep us pinned at `"^0.7.30"`

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, thank you for the steps! I was able to get it working but found the dependency is still using 0.7.18.
Is there a way to change this or will it be okay to leave it how it i
dependency
s?

**Comment by bryceosterhaus** [comment]
@fortunatomaldonado you can inspect the node_modules folder to see what the actual package is, but I believe that yarn resolves that dependency to the newer version.

image

I read this as "ua-parser-js versions `^0.7.18` and `^0.7.30` resolve to `0.7.31`"

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, thank you for the clarification. I updated the commit. Let me know if there is any other things to do.
Thank you!

File: modules/yarn.lock [hunk 1]
```diff
@@ -12856,6 +12856,16 @@ react-text-mask@5.4.3:
...
+react-transition-group@4.4.1, react-transition-group@^4.4.1:
+  version "4.4.1"
+  resolved "https://registry.yarnpkg.com/react-transition-group/-/react-transition-group-4.4.1.tgz#63868f9325a38ea5ee9535d828327f85773345c9"
+  integrity sha512-Djqr7OQ2aPUiYurhPalTrVy9ddmFCCzwhqQmtN+J3+3DzLO209Fdr70QrN8Z3DsglWql6iY1lDWAfpFiBtuKGw==
+  dependencies:
+    "@babel/runtime" "^7.5.5"
+    dom-helpers "^5.0.1"
+    loose-envify "^1.4.0"
+    prop-types "^15.6.2"
+
```
Comment (bryceosterhaus) [inline modules/yarn.lock:12868]: Hmm, this change seems odd. You might try re-generating the yarn.lock by discarding the chages and then re-running `yarn install`

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, okay I think I got it now. Thank you.

**Comment by bryceosterhaus** [comment]
ci:forward

**Comment by bryceosterhaus** [comment]
thanks @fortunatomaldonado!

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
whoops, wrong workflow for this repo...

**Comment by bryceosterhaus** [comment]
I sent it to brian directly since I accidentally merged and was unable to forward https://github.com/brianchandotcom/liferay-portal/pull/109792

**Comment by fortunatomaldonado** [comment]
@bryceosterhaus, thank you so much for all your help!

**Pull request merged**